docs/apache-airflow/public-airflow-api.rst

```diff
@@ -0,0 +1,141 @@
+and should instead use other available methods for interacting with the system.
+
+Similarly, :doc:`Database structure <database-erd-ref>`_ is considered to be an internal implementation
+detail of Apache Airflow and you should not assume the structure is going to be maintained in
+backwards-compatible way.
+
+Is Airflow DB part of the Public API of Airflow?
+================================================
+
+The :doc:`Airflow DB <database-erd-ref>`_ is not considered to be part of the public API of Apache Airflow.
+The Airflow DB is the internal database used by Airflow to store information about DAGs, tasks, and
+other aspects of the system's state. This database is not intended to be accessed directly by
+external developers, and the specific details of its schema and structure are not considered
+to be part of the public API.
+
+Instead, developers who want to interact with the Airflow DB should use the :doc:`stable-rest-api-ref` and
+the :doc:`stable-cli-ref`, both providing a number of different interfaces and methods for accessing and
+manipulating data in the database. These APIs are designed to be more stable and user-friendly than
+accessing the Airflow DB directly, and provide a more consistent and predictable way of interacting
+with the system.
+
+Is Stable REST API part of the Public API of Airflow?
+=====================================================
+
+The :doc:`Stable REST API <stable-cli-ref>`_ is considered to be part of the public API of Apache Airflow.
+The REST API is a set of programming interfaces that allow developers to access certain features of
+Airflow over the HTTP protocol, using a standard set of RESTful (representational state transfer) conventions.
+
+The stable REST API is considered to be part of the public API because it is designed to be used by
+external developers to interact with Airflow in a consistent and predictable way. The REST API provides
+access to many of the same features and functionality that are available through the other public APIs
+of Airflow, such as the ability to create and manage DAGs and tasks, and to monitor their status.
+
+Overall, the stable REST API is an important part of the public API of Apache Airflow, and can
+be a useful tool for developers who want to integrate Airflow with other systems or
+build custom tools and applications on top of Airflow.
+
+Which classes and packages are a public API of Apache Airflow?
+==============================================================
+
+The specific classes and packages that are considered to be part of the public API of Apache
+Airflow can vary depending on the version of Airflow that you are using. In general, however, the
+public API of Airflow consists of a number of different classes and packages that provide access
+to the core features and functionality of the system.
+
+Some examples of the classes and packages that may be considered as the public API of Apache Airflow include:
```

Comment:
Is it not possible to create a complete list? I think the list of interfaces remains fairly static over versions.

Reply:
yes. I removed the examples and made it more complete (looking also for more classes that should be added here).

Eventually we might want to automate control over it but we need to start with documented list.